### PR TITLE
ADM-831:[frontend]feat: add future time check for metrics step

### PR DESCRIPTION
**Pull request by weiraneve**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: frontend/src/containers/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelection/index.tsx [hunk 1]
```diff
@@ -137,6 +141,8 @@ export const PipelineMetricSelection = ({
           options={stepsOptions}
           label={'Step'}
           value={validStepValue}
+          isError={isFutureTime}
+          errorText={NO_PIPELINE_STEP_ERROR}
```
Comment (guzhongren) [inline frontend/src/containers/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelection/index.tsx:145]: Write UT to cover it

**Comment by guzhongren** [comment]
https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&id=au-heartbeat-heartbeat-frontend&open=AY4MokmNBzss7mO_gya9
@weiraneve

**Comment by weiraneve** [comment]
> https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&id=au-heartbeat-heartbeat-frontend&open=AY4MokmNBzss7mO_gya9 @weiraneve

In my ADM-831 branch, this has not appeared again

File: frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelection.test.tsx [hunk 2]
```diff
@@ -137,6 +140,28 @@ describe('PipelineMetricSelection', () => {
     expect(mockHandleClickRemoveButton).toHaveBeenCalledWith(mockId);
   });
 
+  it('should render pipeline no step error given now is 2024-01-01 when get pipeline when selected past time ', async () => {
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/DeploymentFrequencySettings/PipelineMetricSelection.test.tsx:143]: 描述和内容对不上

File: frontend/__tests__/containers/MetricsStep/MetricsStep.test.tsx [hunk 3]
```diff
@@ -304,49 +304,6 @@ describe('MetricsStep', () => {
...
-    });
-
-    it('should render pipeline container given now is 2024-01-01 when get pipeline when selected past time ', async () => {
-      const MOCK_PAST_DATE_RANGE = {
-        startDate: '2014-04-04T00:00:00+08:00',
-        endDate: '2014-04-18T00:00:00+08:00',
-      };
-
-      jest.useFakeTimers().setSystemTime(new Date('2024-01-01'));
-
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/MetricsStep.test.tsx:338]: 漏了一个测试？

**Review by neomgb — APPROVED** [review]

**Pull request merged**